app/src/main/java/com/x8bit/bitwarden/ui/platform/components/snackbar/BitwardenSnackbarHostState.kt

```diff
@@ -0,0 +1,66 @@
+package com.x8bit.bitwarden.ui.platform.components.snackbar
+
+import androidx.compose.material3.SnackbarDuration
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.SnackbarResult
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.remember
+import com.x8bit.bitwarden.ui.platform.base.util.Text
+
+/**
+ * A custom state holder for [BitwardenSnackbarData] and manging a snackbar host with the
+ * passed in [SnackbarHostState].
+ */
+@Stable
+class BitwardenSnackbarHostState(
+    val snackbarHostState: SnackbarHostState,
+) {
+    /**
+     * The current snackbar data to be displayed.
+     */
+    var currentSnackbarData: BitwardenSnackbarData? = null
+        private set
+
+    /**
+     * Shows a snackbar with the given [snackbarData]. Passes the [BitwardenSnackbarData.key]
+     * through the message parameter of the [SnackbarHostState.showSnackbar] method. This key
+     * can be used to identify the correct snackbar data to show in the host.
+     */
+    suspend fun showSnackbar(
+        snackbarData: BitwardenSnackbarData,
+        duration: SnackbarDuration = SnackbarDuration.Short,
+    ): SnackbarResult {
+        currentSnackbarData = snackbarData
+        return snackbarHostState.showSnackbar(message = snackbarData.key, duration = duration)
```

Comment:
Do we need anything to clear the `currentSnackbarData`. I am looking at how the regular Snackbar does it and it looks like this:

```kotlin
    suspend fun showSnackbar(visuals: SnackbarVisuals): SnackbarResult =
        mutex.withLock {
            try {
                return suspendCancellableCoroutine { continuation ->
                    currentSnackbarData = SnackbarDataImpl(visuals, continuation)
                }
            } finally {
                currentSnackbarData = null
            }
        }
```

Reply:
we can add something similar, since the snackbar data in the wrapped M3 snackbar host state is doing this and showing the custom one is dependent on this.